lib/cachex/services/courier.ex

```diff
@@ -78,6 +89,7 @@ defmodule Cachex.Services.Courier do
               spawn_link(fn ->
                 result =
                   try do
+                    put_callers(callers)
```

Comment:
Continuing https://github.com/whitfin/cachex/pull/407#discussion_r2043016067

I would much rather have a solid chain, because even though it is generally used for mocks and the mocks don't care, it's _only_ useful for mocks if it's not keeping the chain properly.

I don't remember why, but there is a specific reason I do not use `Task` here. I believe in older Elixir versions it was not behaving as it should, so I opted to `spawn_link/1` instead. No reason to change it just for this :)

Reply:
Yep, I agree, I wouldn't change stuff that works just for that :) I'll add the courier process itself to the `$callers` chain.